packages/labs/src/select.tsx

```diff
@@ -0,0 +1,121 @@
+
+import {
+    Button,
+    Classes,
+    HTMLInputProps,
+    IInputGroupProps,
+    InputGroup,
+    IPopoverProps,
+    Popover,
+    Position,
+} from "@blueprintjs/core";
+import { ICoreInputListProps, IInputListRenderProps, InputList } from "./inputList";
+
+// a quick SFC to conditionally render InputGroup "X" button
+const InputClearButton: React.SFC<{ onClick: React.MouseEventHandler<HTMLElement>, visible: boolean }>
+    = ({ onClick, visible }) => {
+        return visible ? <Button className={Classes.MINIMAL} iconName="cross" onClick={onClick} /> : null;
+    };
+
+export interface ISelectProps<D> extends ICoreInputListProps<D> {
+    /**
+     * Whether the dropdown list can be filtered.
+     * Disabling this option will remove the `InputGroup` and ignore `inputProps`.
+     * @default true
+     */
+    filterable?: boolean;
+
+    /**
+     * Custom renderer for an item in the dropdown list. Receives a boolean indicating whether
+     * this item is active (selected by keyboard arrows) and an `onClick` event handler that
+     * should be attached to the returned element.
+     */
+    itemRenderer: (item: D, isActive: boolean, onClick: React.MouseEventHandler<HTMLElement>) => JSX.Element;
+
+    /** React child to render when filtering items returns zero results. */
+    noResults?: string | JSX.Element;
+
+    /** React props to spread to `InputGroup`. All props are supported except `ref` (use `inputRef` instead). */
+    inputProps?: IInputGroupProps & HTMLInputProps;
+
+    /** React props to spread to `Popover`. Note that `content` cannot be changed. */
+    popoverProps?: Partial<IPopoverProps>;
+}
+
+@PureRender
+export class Select<D> extends React.Component<ISelectProps<D>, {}> {
+    public static displayName = "Blueprint.Select";
+    public static defaultProps: Partial<ISelectProps<any>> = {
+        filterable: true,
+        inputProps: {},
+        popoverProps: {},
+    };
+
+    public static ofType<T>() { return Select as new () => Select<T>; }
+
+    private DInputList = InputList.ofType<D>();
+
+    public render() {
+        // omit props specific to this component, spread the rest.
+        // TODO: should InputList just support arbitrary props? could be useful for re-rendering
+        const { filterable, itemRenderer, inputProps, noResults, popoverProps, ...props } = this.props;
+        return <this.DInputList renderer={this.renderInputList} {...props} />;
+    }
+
+    private renderInputList = (listProps: IInputListRenderProps<D>) => {
+        const { onKeyDown, onQueryChange, query } = listProps;
+
+        const { ref, ...htmlInputProps } = this.props.inputProps;
+        const input = (
+            <InputGroup
+                autoFocus={true}
+                leftIconName="search"
+                onChange={onQueryChange}
+                placeholder="Filter..."
+                rightElement={<InputClearButton onClick={this.clearQuery} visible={query.length > 0} />}
```

Comment:
it fits on one line and doesn't require a `const`. analogous to `renderItems` member below; is that pattern preferable?